SkietbaanBE/SkietbaanBE/Controllers/UserController.cs

```diff
@@ -60,46 +60,35 @@ public async Task<IActionResult> AddUser([FromBody] User user)
                 return new BadRequestObjectResult("user cannot be null");
             }
         }
-        // PUT: api/User/
+        // PUT: api/User
         [HttpPut("{id}")]
         public async Task<IActionResult> UpdateUser(int id, [FromBody] User user)
         {
-            if (ModelState.IsValid)
-            {
                 //error handling, check if client provided valid data
                 if (user == null)
                 {
                     return new BadRequestObjectResult("user cannot be null");
                 }
                 else
                 {
-                    User dbUser = null; //assume user does not exist
-                    using (_context)
+                    User dbUser = _context.Users.Where(u => u.Username == user.Username && u.Id != user.Id) //check if a different user with the new username already exists
```

Comment:
I would change this to be a boolean with an `Any` query. This is a new username, so you only care about the first occurrence of this username. Your current statement will iterate all the records regardless if it has been found or not. Using an Any() will stop the search once a matching username was found, increasing the performance.